AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/holder/EComponentWithViewSupportHolder.java

```diff
@@ -115,7 +119,12 @@ public void assignFindViewById(JFieldRef idRef, JClass viewClass, JFieldRef fiel
 		} else {
 			assignExpression = findViewById(idRef);
 			if (viewClass != null && viewClass != classes().VIEW) {
-				assignExpression = cast(viewClass, assignExpression);
+				JClass targetClass = codeModelHelper.typeMirrorToJClass(annotatedElement.asType(), this);
```

Comment:
You could use this code (`codeModelHelper.typeMirrorToJClass(annotatedElement.asType(), this)`) in `ViewByIdHandler` so that you don't have to pass the annotatedElement.